torch/_subclasses/meta_utils.py

```diff
@@ -280,10 +361,13 @@ def __call__(self, t, shape_env=None):
                 # tests all break so we just exclude this.  In any case
                 # the to conversion isn't really right anyhow.
                 self.miss += 1
+                if strict:
+                    return NotImplemented
                 return t
             else:
                 self.hit += 1
-                r = self.meta_tensor(t, shape_env=shape_env)
+                r = self.meta_tensor(t, shape_env=shape_env, callback=callback)
+                # TODO: this is suspicious, now that we have callback argument
                 if type(t) is torch.nn.Parameter:
```

Comment:
this is suspicious because the callback could call `torch.nn.Parameter` on the output instead of here ?

Reply:
yeah, the whole point of the callback argument was to make sure we preserved non-leafness, but now we've gone and wrapped it in another Tensor constructor after we processed everything. But maybe it's ok, cuz parameters are always leafs anyway??